### ISSUE TITLE
gettimeofday() does not work

### ISSUE THREAD

**Issue by Hermann-SW** [raspberrypi/pico-sdk#27]
struct timeval seconds are always 0, regardless how much time passed (also for sleep_ms(3333)):
```
tud_cdc_connected()                                    
0 / 536877456                                          
0 / 3758157056                                         
0 / 536873932

```

Code:
```
#include <stdio.h>
#include "pico/stdlib.h"
#include <tusb.h>
#include <sys/time.h>

int main(void)
{
  stdio_init_all();
  while (!tud_cdc_connected()) { sleep_ms(100);  }
  printf("tud_cdc_connected()\n");

  struct timeval tv0,tv1,tv2;
  gettimeofday(&tv0, NULL);
  sleep_ms(1);
  gettimeofday(&tv1, NULL);
  sleep_ms(2);
  gettimeofday(&tv2, NULL);

  printf("%ld / %u\n",tv0.tv_sec,tv0.tv_usec);
  printf("%ld / %u\n",tv1.tv_sec,tv1.tv_usec);
  printf("%ld / %u\n",tv2.tv_sec,tv2.tv_usec);

  return 0;
}

```

**Comment by kilograham** [comment]
Note that this is actually a newlib issue.

It does return -1 for an error, and sets errno to ENOSYS (i.e. not implemented)

**Comment by kilograham** [comment]
won't fix

**Comment by kilograham** [comment]
this is trivial to fix, so may as well

**Comment by kilograham** [comment]
moving to 1.3.0 with other newlib stuff

**Comment by kilograham** [comment]
fixed by #1186

**Comment by kilograham** [comment]
merged into develop
